src/codemodder/codemods/utils_mixin.py

```diff
@@ -58,6 +63,44 @@ def _is_direct_call_from_imported_module(
                             return (import_node, alias)
         return None
 
+    def get_imported_prefix(
+        self, node
+    ) -> Optional[tuple[Union[cst.Import, cst.ImportFrom], cst.ImportAlias]]:
+        """
+        Given a node representing an access, finds if any part of its prefix is imported.
+        Returns a import and import alias pair.
+        """
+        for nodo in iterate_left_expressions(node):
+            match nodo:
+                case cst.Name() | cst.Attribute():
+                    maybe_assignment = self.find_single_assignment(nodo)
+                    if maybe_assignment and isinstance(
+                        maybe_assignment, ImportAssignment
+                    ):
+                        import_node = maybe_assignment.node
+                        for alias in import_node.names:
+                            if maybe_assignment.name in (
+                                alias.evaluated_alias,
+                                alias.evaluated_name,
+                            ):
+                                return (import_node, alias)
+        return None
+
+    def get_aliased_prefix_name(self, node: cst.CSTNode, name: str):
+        """
+        Returns the alias of name if name is imported and used as a prefix for this node.
+        """
+        maybe_import = self.get_imported_prefix(node)
+        maybe_name = None
+        if maybe_import:
+            imp, ia = maybe_import
+            match imp:
+                case cst.Import():
```

Comment:
I love match/case style but I Think in this particular case maybe an if statemtn or walrus is best since it's less code indentation.

Reply:
You're right here, but using match/case has the advantage of signaling the static analysis tools about the type.